docs/docs-config.d.ts

```diff
@@ -0,0 +1,11 @@
+declare module 'docs/config' {
```

Comment:
@brijeshb42 moving the conversation here.

Do you remember why we renamed the file to be a JS file [here](https://github.com/mui/material-ui/pull/46834/files#diff-cc4a2fc39fb7fe6eea3d4f1452e1999a16e937c1ab9d038054735f2a456f57e6)? If we revert it to TS then we wouldn't need to keep file this up to date.